metricflow/time/time_granularity_solver.py

```diff
@@ -116,35 +128,75 @@ def resolve_granularity_for_partial_time_dimension_specs(
         return result
 
     def find_minimum_granularity_for_partial_time_dimension_spec(
-        self, partial_time_dimension_spec: PartialTimeDimensionSpec, metric_references: Sequence[MetricReference]
+        self,
+        partial_time_dimension_spec: PartialTimeDimensionSpec,
+        metric_references: Sequence[MetricReference],
+        read_nodes: Sequence[BaseOutput],
+        node_output_resolver: DataflowPlanNodeOutputDataSetResolver,
```

Comment:
Ok. I think it might be better to keep this totally separate from the new function and branch at the callsite for now. 

There are only two callsites, one of which is in this class, and that way we aren't tempted to keep adding responsibility to the node output resolver.

What do you think? Otherwise it might be worth it to mark the new method with the leading _ for the time being, I'm not sure we'll ever really want to call it on its own if we keep this method signature.

Reply:
Not totally sure I understand what you mean. Is it this? We don't want to use the `node_output_resolver` here, so we should pass in the output datasets already resolved.

I'm thinking maybe it would be better to do this all on init for this class. Iterate through all the datasets and store a dict of all `time_dimension_instances` to a set of `granularities` (currently should only be one per instance). Then it will be super easy to grab that at query time.

Reply:
^ implemented it this way!